shell/browser/serial/electron_serial_delegate.cc

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "shell/browser/serial/electron_serial_delegate.h"
+
+#include <utility>
+
+#include "content/public/browser/web_contents.h"
+#include "shell/browser/serial/serial_chooser_context.h"
+#include "shell/browser/serial/serial_chooser_context_factory.h"
+#include "shell/browser/serial/serial_chooser_controller.h"
+
+namespace electron {
+
+SerialChooserContext* GetChooserContext(content::RenderFrameHost* frame) {
+  auto* web_contents = content::WebContents::FromRenderFrameHost(frame);
+  auto* browser_context = web_contents->GetBrowserContext();
+  return SerialChooserContextFactory::GetForBrowserContext(browser_context);
+}
+
+ElectronSerialDelegate::ElectronSerialDelegate() = default;
+
+ElectronSerialDelegate::~ElectronSerialDelegate() = default;
+
+std::unique_ptr<content::SerialChooser> ElectronSerialDelegate::RunChooser(
+    content::RenderFrameHost* frame,
+    std::vector<blink::mojom::SerialPortFilterPtr> filters,
+    content::SerialChooser::Callback callback) {
+  chooser_controller_ = std::make_unique<SerialChooserController>(
+      frame, std::move(filters), std::move(callback));
+  return nullptr;
+}
+
+bool ElectronSerialDelegate::CanRequestPortPermission(
+    content::RenderFrameHost* frame) {
+  return true;
```

Comment:
Should this be sent through to the permission check handler? https://www.electronjs.org/docs/api/session#sessetpermissioncheckhandlerhandler

Reply:
Yes, and it should default to rejected

Reply:
👍 ok I'll do so.

Reply:
@nornagon @MarshallOfSound web serial uses [ContentSettingsType::SERIAL_GUARD](https://source.chromium.org/chromium/chromium/src/+/master:chrome/browser/serial/serial_chooser_context.cc;l=66) for permissioning, but it doesn't look like there is a corresponding [content::PermissionType](https://source.chromium.org/chromium/chromium/src/+/master:components/permissions/permission_util.cc;l=128).  Any suggestions with what we do with that?

Reply:
@MarshallOfSound FWIW the way https://www.electronjs.org/docs/api/session#sessetpermissioncheckhandlerhandler works it defaults to accepted unless you add a handler.